hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1386,6 +1387,14 @@ bool MDMParser::registerNet(const char* apn, NetStatus* status, system_tick_t ti
                         break; // force another checkNetStatus() call
                     }
                 }
+                // Query IMSI every 60 sec if not regsitered, to detect IMSI switches
+                if (HAL_Timer_Get_Milli_Seconds() - start_imsi_check > 60000UL) {
+                    start_imsi_check = HAL_Timer_Get_Milli_Seconds();
+                    sendFormated("AT+CIMI\r\n");
+                    if (RESP_OK != waitFinalResp()) {
+                        goto failure;
+                    }
+                }
```

Comment:
please check this? I didn't do a `CIMI_TIMEOUT` because it takes default 10 sec...